tests/estimator/test_resources_base.py

```diff
@@ -0,0 +1,279 @@
+        """Test that the repr method correctly represents the class."""
+        resources = Resources(wire_manager=wm, gate_types=gt)
+
+        expected_wm = wm
+        expected_gt = defaultdict(int, {}) if gt is None else gt
+        assert repr(resources) == f"Resources(wire_manager={expected_wm}, gate_types={expected_gt})"
+
+    def test_gate_counts(self):
+        """Test that this function correctly simplifies the gate types
+        dictionary by grouping together gates with the same name."""
+
+        class DummyResOp2:
+            """A dummy class to populate the gate types dictionary for testing."""
+
+            def __init__(self, name, parameter=None):
+                """Initialize dummy class."""
+                self.name = name
+                self.parameter = parameter
+
+            def __hash__(self):
+                """Custom hash which only depends on instance name."""
+                return hash((self.name, self.parameter))
+
+        rx1 = DummyResOp2("RX", parameter=3.14)
+        rx2 = DummyResOp2("RX", parameter=3.14 / 2)
+        cnots = DummyResOp2("CNOT")
+        ry1 = DummyResOp2("RY", parameter=3.14)
+        ry2 = DummyResOp2("RY", parameter=3.14 / 4)
+
+        gate_types = {rx1: 1, ry1: 2, cnots: 3, rx2: 4, ry2: 5}
+        res = Resources(wire_manager=wm1, gate_types=gate_types)
+
+        expected_gate_counts = {"RX": 5, "RY": 7, "CNOT": 3}
+        assert res.gate_counts == expected_gate_counts
+
+    def test_equality(self):
+        """Test that the equality method works as expected."""
+        gt1, gt2 = (gate_types_data[0], gate_types_data[1])
+
+        res1 = Resources(wire_manager=wm1, gate_types=gt1)
+        res1_copy = Resources(wire_manager=wm1, gate_types=gt1)
+        res2 = Resources(wire_manager=wm2, gate_types=gt2)
+
+        assert res1 == res1
+        assert res1 == res1_copy
+        assert res1 != res2
+
+    def test_arithmetic_raises_error(self):
+        """Test that an assertion error is raised when arithmetic methods are used"""
+        res = Resources(wire_manager=wm1, gate_types=gate_types_data[0])
+
+        with pytest.raises(AssertionError):
+            _ = res + 2  # Can only add two Resources instances
+
+        with pytest.raises(AssertionError):
+            _ = res & 2  # Can only add two Resources instances
+
+        with pytest.raises(AssertionError):
+            _ = res * res  # Can only multiply a Resources instance with an int
+
+        with pytest.raises(AssertionError):
+            _ = res @ res  # Can only multiply a Resources instance with an int
+
+    def test_add_in_series(self):
+        """Test that we can add two resources assuming the gates occur in series"""
+        res1 = Resources(wire_manager=wm3, gate_types=gate_types_data[2])
+        res2 = Resources(wire_manager=wm2, gate_types=gate_types_data[1])
+
+        expected_wm_add = WireResourceManager(
+            clean=8753,  # max(clean1, clean2)
```

Comment:
I would personally say this is better testing since it is testing the behavioural logic rather than a static magic number.
```suggestion
            clean=max(res1.wire_manager.clean, res2.wire_manager.clean),  # max(clean1, clean2)
```
I don't think this suggestion will work automatically but do you mind updating the entire test file to follow this philosophy?